server/api/users/users.controller.ts

```diff
@@ -46,3 +54,34 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 
   res.json(user);
 };
+
+export const updateUserById: RequestHandler<{ id: string }, string, UpdatableFields> = async (req, res) => {
+
+  const acceptableFields = [
+    'bio',
+    'twitter_username',
+    'linkedin_url',
+    'github_username',
+    'website',
+  ];
+
+  const toUpdate: UpdatableFields = {};
+
+  acceptableFields.forEach((field) => {
+    if (field in req.body) {
+      toUpdate[field as keyof UpdatableFields] = req.body[field as keyof UpdatableFields];
+    }
+  });
+
+  const updated = await User.update(toUpdate, {
```

Comment:
How will this handle an empty object? Should a query be executed at all if no fields have changed?

Reply:
That's a great question! A query probably shouldn't be executed. I've added a check and test for this!